### ISSUE TITLE
Make the admin page to recommend caching WP core components and theme well known assets

### ISSUE THREAD

**Issue by delapuente** [darkwing/wp-sw-cache#18]
The admin page should recommend core components and CSS, fonts, scripts and images of the theme to cache by checking them by default while keeping all non recognized files as opt-in.

**Comment by marco-c** [comment]
Duplicate of #16?

**Comment by delapuente** [comment]
Nope. This is a UX proposal to check core components and well known assets by default. The other is including core components in the list (checked or not).

**Comment by darkwing** [comment]
@digitarald mentioned we could curl the homepage and then parse the response for images, css, and JavaScript files.

**Comment by darkwing** [comment]
I don't see this as being "comprehensive" for V1 but more logic would be nice as MVP.

**Comment by marco-c** [comment]
> @digitarald mentioned we could curl the homepage and then parse the response for images, css, and JavaScript files.

I think this isn't feasible. If the homepage is a list of posts, we would cache some unneeded resources (e.g. the images that are in those posts). If the homepage is a page, we could miss caching some resources (e.g. if the theme has different layouts for pages and posts). It doesn't seem much better than the current heuristic + #16.

Maybe, if we merged your plugins, we could cache things lazily and avoid the server-side logic. We could make the service worker request things from the network and then add them to the cache (this strategy wouldn't work if you only want to cache the shell, but would work if you want to cache everything).

**Comment by darkwing** [comment]
I got an awesome tip that we need to be careful about the dynamic caching stuff; someone could request a bunch of pages with different querystring values and load up peoples' cache with garbage.

I think we're stuck right now between two minds:  

1. The WordPress admin needs to know what's happening and should pick files to cache explicitly (though we can create some logic to help them out)

2. The WordPress admin should know nothing and the plugin should be "plug and play".

I believe in option 1 as an MVP (version 1); a bad SW could really cause problems so it's important the dev know what's going on.  Option 2 is a good long term goal though.

**Comment by marco-c** [comment]
I agree with you, although I think many users will just go with whatever the plugin suggests, so in a way option 1 is often the same as option 2 and we should be extra careful anyway.
